.github/workflows/cmake.yml

```diff
@@ -325,10 +326,15 @@ jobs:
         LSAN_OPTIONS: ${{ matrix.lsan-options || 'verbosity=1' }}
 
     - name: Upload coverage report
-      if: matrix.codecov
+      if: matrix.codecov && ( env.CODECOV_TOKEN_SECRET != '' || github.repository == 'zlib-ng/zlib-ng' )
+      shell: bash
       run: |
-        cd ${{ matrix.build-dir || '.' }}
-        python -m codecov -F "${{ matrix.codecov }}" --name "${{ matrix.name }}" --gcov-exec="${{ matrix.gcov-exec || 'gcov' }}"
+        bash tools/codecov-upload.sh
       env:
         # Codecov does not yet support GitHub Actions
-        CODECOV_TOKEN: "${{secrets.CODECOV_TOKEN}}"
\ No newline at end of file
+        CODECOV_TOKEN_SECRET: "${{secrets.CODECOV_TOKEN}}"
```

Comment:
Is `CODECOV_TOKEN_SECRET` needed?

Reply:
It is needed for pull requests as the scope for them is the source repository not zlib-ng/zlib-ng which is the destination repository.

Reply:
The variable is not used in the script.

Reply:
See lines 329 and 336... It uses the hard-coded token for zlib-ng/zlib-ng repository but allows overriding it for other repositories...

Reply:
Couldn't an overriding repository just set `secrets.CODECOV_TOKEN` in their repository? That is what I do in mine.

Reply:
Yes... But the action still wouldn't have access to the variable... It's basically security restriction...

Reply:
You mean the script wouldn't have access to that variable? It has access to `CODECOV_TOKEN` already, so there is no need for a duplicate token `CODECOV_TOKEN_SECRET`. The action does have access to `secrets.CODECOV_TOKEN`. The `if` statement above does not work in my repository. I had to modify it. There is a mismatch between an environment variable and a secrets variable I think. There is no need for an environment variable because everything is run by a GitHub Actions runner which has secrets variables.

Reply:
It doesn't have access to the variable when it's a pull request... There is no problems when it's normal commit.

Reply:
I think that is why it is checking if repo is zlib-ng in the `if` statement for PR and the || statement in `CODECOV_TOKEN`.

Reply:
My knowledge of codecov is quite limited, so I don't know how to fix it so it works 100%... I just know the limitations...

Reply:
These things are difficult for me to test, because I have access. A PR from someone external could experiment with whether this is actually needed or not, but as-is should work for everyone. My testing led me to believe that this was needed, but it might possibly not be.

Reply:
https://github.community/t5/GitHub-Actions/jobs-lt-job-id-gt-if-does-not-work-with-env-secrets/td-p/38549
https://github.community/t5/GitHub-Actions/If-expression-with-context-variable/td-p/34556

There doesn't however appear to be a way for me to even set special `env` vars, so using it is not an option.